react/LocationSearch.tsx

```diff
@@ -1,37 +1,60 @@
-import React, { useState, useMemo } from 'react'
+import React, { useState, useMemo, useRef } from 'react'
 import { FormattedMessage } from 'react-intl'
-import { Input, IconSearch, IconClear } from 'vtex.styleguide'
+import { Input, IconSearch, IconClear, IconWarning } from 'vtex.styleguide'
+import { positionMatchWidth } from '@reach/popover'
+
 import {
   Combobox,
   ComboboxInput,
   ComboboxOption,
   ComboboxPopover,
   ComboboxList,
-} from '@reach/combobox'
-import '@reach/combobox/styles.css'
-
-import { addresses as mockedAddresses } from './addresses'
+} from './components/Combobox'
 import styles from './LocationSearch.css'
+import PlaceIcon from './components/PlaceIcon'
 
-const MAX_DROPDOWN_ADDRESSES = 6
+interface Interval {
+  offset: number
+  size: number
+}
 
-// This function will be replaced in the future, after integrating the
-// component with GraphQL queries.
-const getAddresses = (searchTerm: string) => {
-  return mockedAddresses
-    .filter((address: string) =>
-      address.toLocaleLowerCase().includes(searchTerm.toLocaleLowerCase())
-    )
-    .slice(0, MAX_DROPDOWN_ADDRESSES)
+export interface Suggestion {
+  description: string
+  mainText: string
+  mainTextMatchInterval: Interval
+  secondaryText: string
+}
+
+const renderSuggestionText = (suggestion: Suggestion) => {
+  const { mainText } = suggestion
+  const { offset, size } = suggestion.mainTextMatchInterval
+  return (
+    <div className="truncate c-muted-2">
```

Comment:
When the text inside an Option is too big, it will truncate with an ellipsis, but there is a problem: the `mainText` (street name) is black and the `secondaryText` (city and state name) is gray, which would require the ellipsis color to be dynamic, depending on which part have overflowed.

Doing this at implementation level is quite painful and don't seems to bring too much value (big street names aren't very common, I guess), so I've opted to always display the ellipsis with a gray color. Is that okay? @davicosta99 @augustocb

Example of fake big street name with the grey colored ellipsis:
![image](https://user-images.githubusercontent.com/26108090/88686971-08efe000-d0ce-11ea-982c-88023a862772.png)

Reply:
I think it's fine, Wesley, thanks for the concern. Appreciate the level of care 😉